File: backend/src/main/java/com/keeper/goal/GoalService.java
package com.keeper.goal;

import com.keeper.dto.*;
import java.util.*;

public interface GoalService {
  GoalDTO createGoal(UUID workspaceId, GoalDTO goalDTO);

  void deleteGoal(UUID workspaceId, UUID id);

  java.util.List<GoalDTO> getGoals(UUID workspaceId, String status);

  GoalDTO updateGoal(UUID workspaceId, UUID id, GoalDTO goalDTO);
}

File: frontend/src/features/goals/api/GoalService.ts
import { GoalApi } from '../../../api-client';
import { apiClient } from '../../../utils/api';

const api = new GoalApi(undefined, '', apiClient);

export const GoalService = {
  getAll: async (workspaceId: string) => {
    const response = await api.getGoals(workspaceId);
    return response.data;
  },
  create: async (workspaceId: string, goal: any) => { 
    const response = await api.createGoal(workspaceId, goal);
    return response.data;
  },
  update: async (workspaceId: string, goalId: string, goal: any) => { 
    const response = await api.updateGoal(workspaceId, goalId, goal);
    return response.data;
  },
  delete: async (workspaceId: string, goalId: string) => {
    await api.deleteGoal(workspaceId, goalId);
  },
  fund: async (workspaceId: string, goalId: string, payload: { amount: string, sourceAccountId: string }) => {
    const response = await apiClient.post(`/api/v1/goals/${workspaceId}/${goalId}/fund`, payload);
    return response.data;
  }
};
